history/src/main/scala/org/apache/spark/deploy/history/SimpleFsHistoryProvider.scala

```diff
@@ -0,0 +1,279 @@
+  }
+
+  override def getAppUI(appId: String, attemptId: Option[String]): Option[LoadedAppUI] = {
+    getApplicationInfoWrapper(appId).flatMap(appInfo => {
+      appInfo.attempts.find(_.info.attemptId == attemptId).map(attemptInfo => {
+        val conf = this.conf.clone()
+        val secManager = new SecurityManager(conf)
+        secManager.setAcls(HISTORY_UI_ACLS_ENABLE)
+        secManager.setAdminAcls(attemptInfo.adminAcls.getOrElse(""))
+        secManager.setViewAcls(attemptInfo.info.sparkUser, attemptInfo.adminAclsGroups.getOrElse(""))
+        secManager.setAdminAclsGroups(attemptInfo.adminAclsGroups.getOrElse(""))
+        secManager.setViewAclsGroups(attemptInfo.viewAclsGroups.getOrElse(""))
+
+        val kvstore = createInMemoryStore(attemptInfo)
+        val ui = SparkUI.create(None, new AppStatusStore(kvstore), conf, secManager, appInfo.info.name,
+          HistoryServer.getAttemptURI(appInfo.info.id, attemptInfo.info.attemptId),
+          attemptInfo.info.startTime.getTime(),
+          attemptInfo.info.appSparkVersion
+        )
+
+        LoadedAppUI(ui)
+      })
+    })
+  }
+  
+ 
+  /**
+   * Replays the events in the specified log file on the supplied `ReplayListenerBus`.
+   * `ReplayEventsFilter` determines what events are replayed.
+   */
+  private def replay(
+      eventLog: FileStatus,
+      bus: ReplayListenerBus,
+      eventsFilter: ReplayEventsFilter = SELECT_ALL_FILTER): Unit = {
+    val logPath = eventLog.getPath()
+    val isCompleted = !logPath.getName().endsWith(EventLoggingListener.IN_PROGRESS)
+    logInfo(s"Replaying log path: $logPath")
+    // Note that the eventLog may have *increased* in size since when we grabbed the filestatus,
+    // and when we read the file here.  That is OK -- it may result in an unnecessary refresh
+    // when there is no update, but will not result in missing an update.  We *must* prevent
+    // an error the other way -- if we report a size bigger (ie later) than the file that is
+    // actually read, we may never refresh the app.  FileStatus is guaranteed to be static
+    // after it's created, so we get a file size that is no bigger than what is actually read.
+    Utils.tryWithResource(EventLoggingListener.openEventLog(logPath, fs)) { in =>
+      logInfo("Trying to replay bus: $isCompleted, $eventsFilter")
+      bus.replay(in, logPath.toString, !isCompleted, eventsFilter)
+      logInfo(s"Finished parsing $logPath")
+    }
+  }
+ 
+  override def writeEventLogs(
+      appId: String,
+      attemptId: Option[String],
+      zipStream: ZipOutputStream): Unit = {
+
+    /**
+     * This method compresses the files passed in, and writes the compressed data out into the
+     * [[OutputStream]] passed in. Each file is written as a new [[ZipEntry]] with its name being
+     * the name of the file being compressed.
+     */
+    def zipFileToStream(file: Path, entryName: String, outputStream: ZipOutputStream): Unit = {
+      val fs = file.getFileSystem(hadoopConf)
```

Comment:
It looks as if we already create a FileSystem object above using the same Hadoop configuration, which I believe would mean we use the same FS implementation. Why do we need to make another here?

Reply:
It could be an edge case where the log file path (defined at the top of the class) isn't on the same FS as the file referred to here, in which case it would matter?

The original spark history provider code also does this, so I'd say we leave it in to keep as close to the original as we can:
https://github.com/apache/spark/blob/branch-2.3/core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala#L600